integration/e2e_test.go

```diff
@@ -64,3 +63,152 @@ func TestRolloutHappyCase(t *testing.T) {
 	requireEventuallyPod(t, api, ctx, "mock-zone-b-0", expectReady(), expectVersion("2"))
 	requireEventuallyPod(t, api, ctx, "mock-zone-c-0", expectReady(), expectVersion("2"))
 }
+
+func TestNoDownscale_CanDownscaleUnrelatedResource(t *testing.T) {
```

Comment:
What do you think about testing the self-signed certificate flow as well? I think it's somewhat complex and it's worth testing the few basic cases for it.

Reply:
Actually all tests test the self-signed certificate flow, but I also added an expiration & renewal test as part of https://github.com/grafana/rollout-operator/pull/25/commits/613d0fc4f70f9d51cc6f525294f336bbfb31575b